pylint/constants.py

```diff
@@ -24,7 +30,7 @@
 # The line/node distinction does not apply to fatal errors and reports.
 _SCOPE_EXEMPT = "FR"
 
-MSG_TYPES = {
+MSG_TYPES: Final = {
```

Comment:
Although `mypy` doesn't recognise that because this `dict` is final its values could be seen as `Literal` instead of `str` it does "allow" us to add an ignore later on.
We will be warned if `MSG_TYPES` is being altered and we can thus be sure that any of the values obtained from it is one of the values already in there.

Reply:
@cdce8p Do you know if there is any plans or previous discussion about the concept of final dicts? I saw a similar issue about final frozen sets I think, which would also be beneficial to us.

I can change this to `TypedDict`, but if there are plans to (at some point) support final dicts I think it makes sense to keep it like this as it "better" represents what this actually is.

Reply:
> @cdce8p Do you know if there is any plans or previous discussion about the concept of final dicts? I saw a similar issue about final frozen sets I think, which would also be beneficial to us.
> 
> I can change this to `TypedDict`, but if there are plans to (at some point) support final dicts I think it makes sense to keep it like this as it "better" represents what this actually is.

I don't know of any plans. If `TypedDicts` work, they might be the easiest solution here.